base/reducedim.jl

```diff
@@ -321,8 +321,10 @@ julia> mapreduce(isodd, |, a, dims=1)
 """
 mapreduce(f, op, A::AbstractArrayOrBroadcasted; dims=:, init=_InitialValue()) =
     _mapreduce_dim(f, op, init, A, dims)
-mapreduce(f, op, A::AbstractArrayOrBroadcasted...; kw...) =
-    reduce(op, map(f, A...); kw...)
+function mapreduce(f, op, A::AbstractArrayOrBroadcasted...; kw...)
+    isempty(kw) && return mapreduce(A->f(A...), op, zip(A...))
```

Comment:
Should this allow `mapreduce(+, +, -3:3, -3:3, init=1)` to take the `zip` path?

Reply:
Yes it should. Is there an easy way to do that?

Reply:
I think the only two legal ones are `dims, init`, but am not sure if `init = _InitialValue()` is understood by all the relevant methods. If not, then this would work:
```
function mapreduce(f, op, A::AbstractArrayOrBroadcasted...; dims=:, kw...)
    dims==(:) && return mapreduce(A->f(A...), op, zip(A...); kw...)
    ...
```

Reply:
Last commit needs `; kw...)` -> `; dims=:, kw...)`, I think.